internal/web/server.go

```diff
@@ -525,10 +526,62 @@ func (s *server) handleOpen(ctx context.Context, w http.ResponseWriter, pkg ctrl
 	if err != nil {
 		s.sendToast(w, toast.WithErr(fmt.Errorf("failed to open %v: %w", pkg.GetName(), err)))
 	} else {
+		s.forwardersMutex.Lock()
```

Comment:
Technically, I think this lock should be acquired before trying to read the forwarder the first time (line 519), to prevent Open being called multiple times. Will have to unlock in error case as well though